### DISCUSSION TITLE
RM3 expansion terms

### DISCUSSION THREAD

**Discussion by C01DZin** [castorini/pyserini#1200] [Q&A] [answered]
are there any methods that I could get rm3 expansion terms ?
Like after setting rm3 parameters and inputting a query, could I call a function to get a list of expansion terms that rm3 uses ?

**Comment by lintool** [comment, marked as answer]
Hi @C01DZin - try this:
https://github.com/castorini/pyserini/blob/master/pyserini/search/lucene/_searcher.py#L226

You can print out the original and expanded query.

**Reply by C01DZin** [reply]
I was trying to find any functions that could return the expansion term list , so I could directly use this expansion terms (python list format) but it looks like the only way is to extract it from debug output. Seems little tricky but probably could work.  Thanks a lot.

**Reply by lintool** [reply]
Yes, it's janky, but the functionality isn't currently exposed in Pyserini. To get access to the terms programatically, you'll need to hack Anserini on the Java end, and then connect to Pyserini in Python...

PR welcome ;)

**Comment by ola13** [comment]
Hi @C01DZin, did you have any luck with enabling returning the expansion term list? I'm interested in this functionality as well and might try working on a PR if you haven't started already

**Reply by lintool** [reply]
Hi @ola13 - you have impeccable timing!

I was just working on this: https://github.com/castorini/anserini/pull/1979 - this PR exposes the feature in Anserini. I'll have to write the hooks on the Pyserini end. Coming in a separate PR.

**Reply by ola13** [reply]
Haha, nice, looking forward to trying it out!
